### ISSUE TITLE
Routes with underline on argument don't work

### ISSUE THREAD

**Issue by lucasbento** [pedronauck/micro-router#8]
An example:

This doesn't work:
```js
module.exports = router(
  del('/:my_id', myFunctionWithRoute),
  post('/*', notFoundFunction),
);
```

But this one does:
```js
module.exports = router(
  del('/:myId', myFunctionWithRoute),
  post('/*', notFoundFunction),
);
```

Note the difference between `my_id` and `myId`.

**Comment by pedronauck** [comment]
I'm using [this package](https://github.com/snd/url-pattern) do parse url, I think that it doesn't accept underline routes

**Comment by ernestofreyreg** [comment]
Perhaps its possible to use https://www.npmjs.com/package/path-to-regexp instead?

**Comment by saggiyogesh** [comment]
Indeed, `path-to-regexp` is compatible with express routes also. So current express routes can be reused.

**Comment by efleurine** [comment]
Under the hood, url-pattern is using path-to-regex. 
So I am not sure switching would solve the problem.

May be that is the kind of behavior they really intended.

If you really think it is  a bug @lucasbento then you should open an issue on url-pattern GitHub.

**Comment by jimthedev** [comment]
This is true for periods as well. Here is a failing test showing what happens when you put a period in a param: https://github.com/jimthedev/micro-router/commit/7da3f9da50e46f6c158997db40d7bb795f702c73

**Comment by pedronauck** [comment]
Closed on [release v3.0.0](https://github.com/pedronauck/micro-router/releases/tag/v3.0.0)
